### PR TITLE
Add Smart Bitcoin Cash

### PR DESCRIPTION
**Pull request by zxh0**
## Description

Add Smart Bitcoin Cash (smartBCH for short). It is a sidechain for Bitcoin Cash and has an aim to explore new ideas and unlock possibilities. It is compatible with Ethereum's EVM and Web3 API and provides high throughput for DApps in a fast, secure, and decentralized manner.

## Testing instructions

Test in the same manner as BSC or Polygon was tested.

## Types of changes

* New feature (non-breaking change which adds functionality) 

## Checklist

- [ ] Prefix PR title with `[WIP]` if necessary.
- [x] Add tests to cover changes as needed.
- [x] Update documentation as needed.
- [x] I have read the [guidelines](https://developer.trustwallet.com/wallet-core/newblockchain#integration-criteria) for adding a new blockchain
- [ ] If there is a related Issue, mention it in the description (e.g. Fixes #<issue_number> ).

### CODE REVIEW

**Comment by catenocrypt** [comment]
Closing on grounds of small support at this point.

**Comment by zxh0** [comment]
> Closing on grounds of small support at this point.

Hi @catenocrypt, could you please explain it in more detail? Thank you.

**Comment by catenocrypt** [comment]
https://developer.trustwallet.com/wallet-core/newblockchain#integration-criteria

**Comment by maplesyrupsucker** [comment]
@catenocrypt Would love to see this PR re-opened if possible. Per requirements: 

✅ The blockchain has launched mainnet and has stably run for at least 3 ~ 6 months without major security incidents.
- https://news.bitcoin.com/bch-ecosystem-gains-new-nft-and-dex-platforms-tvl-in-smartbch-defi-climbs-180-in-30-days/
- $26M TVL on smartBCH https://defillama.com/chain/smartBCH

✅ The blockchain has extensive public documentation and tools available for developers to use.
- https://docs.smartbch.org/smartbch/
- https://mainnet.cash/tutorial/smartbch.html
- https://smartbch.fountainhead.cash/mainnet
- https://www.smartscan.cash/
- https://app.mistswap.fi/swap
- https://benswap.cash/

✅ The native coin is listed in the top 30 coins on CoinMarketCap 
- native coin is Bitcoin Cash (Currently 27 on cmc) https://coinmarketcap.com/currencies/bitcoin-cash/

(TBD) - and proposal gets approved on [https://governance.trustwallet.com].

✅ The project needs to provide API/JSON-RPC access to the node with a load balancing setup for private use, see detail requirements [here](https://developer.trustwallet.com/wallet-core/rpc-requirements).
- https://docs.smartbch.org/smartbch/developers-guide/test-using-metamask
- https://docs.smartbch.org/smartbch/developers-guide/jsonrpc

✅ The native coin is tradable on major exchanges.
- https://coinmarketcap.com/currencies/bitcoin-cash/

**Comment by catenocrypt** [comment]
This PR is based on a rather outdated fork (November), and there is also merge conflict compared to master.
Please merge latest master into it.

**Comment by zxh0** [comment]
Hi @catenocrypt , I have updated this PR based on the latest master to get a clean commit. Thank you.

**Comment by catenocrypt** [comment]
@zxh0 , please also extend TWAnyAddress test, see
https://github.com/trustwallet/wallet-core/blob/master/tests/interface/TWAnyAddressTests.cpp#L56

**Comment by zxh0** [comment]
@catenocrypt , added smartBCH testcase into tests/interface/TWAnyAddressTests.cpp

File: docs/registry.md [hunk 1]
```diff
@@ -73,6 +73,7 @@ This list is generated from [./registry.json](../registry.json)
 | 10000070 | Optimistic Ethereum | OETH   | <img src="https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/optimism/info/logo.png" width="32" />     | <https://optimism.io/>        |
 | 10000100 | Gnosis Chain     | xDAI   | <img src="https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/xdai/info/logo.png" width="32" />         | <https://www.xdaichain.com>   |
 | 10000118 | Osmosis          | OSMO   | <img src="https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/osmosis/info/logo.png" width="32" />      | <https://osmosis.zone/>       |
+| 10000145 | Smart Bitcoin Cash | BCH    | <img src="https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/smartbch/info/logo.png" width="32" />     | <https://smartbch.org/>       |
```
Comment (hewigovens) [inline docs/registry.md:76]: Could you please add your logo to https://github.com/trustwallet/assets?

**Review by hewigovens — APPROVED** [review]

File: docs/registry.md (hunk 1 above)
Comment (zxh0) [reply docs/registry.md:76]: Already added: https://github.com/trustwallet/assets/pull/9710

**Review by catenocrypt — APPROVED** [review]

**Pull request merged**